src/Umbraco.Cms.Integrations.SEO.SemrushTools/Umbraco.Cms.Integrations.SEO.SemrushTools/Services/ISemrushService.cs

```diff
@@ -0,0 +1,12 @@
+﻿
```

Comment:
Think we might look at the naming and signatures here.  On the naming, this is perhaps a bit more focussed than `SemrushService` might suggest.  If you compare to the HubSpot one, you'll see that we have a general `HubSpotService`, but that's because much of the equivalent code that you have in your controller - i.e. making the API requests - is pushed up to the service.  Leading to "thinner controllers" which you might argue is a good thing.  However, unless you are minded to, I don't think you need to refactor to that here.

I would suggest though that this is better with a more focussed name, perhaps `ISemrushTokenService` and I don't think the generic parameter is really adding much value, and would look to change to:

```
    public interface ISemrushTokenService
    {
        bool TryGetParameters(string key, out TokenDto obj);
        ...
```